### PR TITLE
int64 support for 'where' op

### PR DESCRIPTION
**Pull request by neginraoof**
Added int64 input type for 'where' op

**Motivation and Context**
This is needed to run the Geminet model. I've uploaded the onnx model here:
https://microsoft-my.sharepoint.com/:f:/p/neraoof/EinycHafN7RIu0PJHQmpFBoBgrnNKagH9THA5XU2-ESC2Q?e=P4ZfSG

Based on the onnx docs (https://github.com/onnx/onnx/blob/master/docs/Operators.md#where), 'where' operator supports all tensor types for input and output.

### CODE REVIEW

File: onnxruntime/core/providers/cpu/cpu_execution_provider.cc [hunk 1]
```diff
@@ -541,6 +542,7 @@ void RegisterOnnxOperatorKernels(KernelRegistry& kernel_registry) {
       BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, string, Where)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, float, Where)>,
       BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, int32_t, Where)>,
+      BuildKernelCreateInfo<ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 9, int64_t, Where)>,
```
Comment (snnn) [inline onnxruntime/core/providers/cpu/cpu_execution_provider.cc]: Did you test it locally?

**Comment by snnn** [comment]
LGTM.
But please also add a test case to 
https://github.com/onnx/onnx/blob/master/onnx/backend/test/case/node/where.py

Now it only contains a float32 test case.

File: onnxruntime/core/providers/cpu/cpu_execution_provider.cc (hunk 1 above)
Comment (neginraoof) [reply onnxruntime/core/providers/cpu/cpu_execution_provider.cc]: Yes, this is tested locally.

**Comment by neginraoof** [comment]
@snnn We would need to merge this before adding the test on ONNX side.
This is the PR though: https://github.com/onnx/onnx/pull/2253

**Pull request merged**